src/main/g8/scalafix/build.sbt

```diff
@@ -14,41 +18,93 @@ inThisBuild(
         url("https://example.com")
       )
     ),
-    addCompilerPlugin(scalafixSemanticdb),
-    scalacOptions ++= List(
-      "-Yrangepos",
-      "-P:semanticdb:synthetics:on"
-    )
+    semanticdbEnabled := true,
+    semanticdbVersion := scalafixSemanticdb.revision
   )
 )
 
-publish / skip := true
+lazy val root = (project in file("."))
+  .aggregate(
+    rules.projectRefs ++
+      input.projectRefs ++
+      output.projectRefs ++
+      tests.projectRefs: _*
+  )
+  .settings(
+    publish / skip := true
+  )
 
-lazy val rules = project.settings(
-  moduleName := "scalafix",
-  libraryDependencies += "ch.epfl.scala" %% "scalafix-core" % V.scalafixVersion
-)
+lazy val rules = projectMatrix
+  .settings(
+    moduleName := "scalafix",
+    libraryDependencies += "ch.epfl.scala" %% "scalafix-core" % V.scalafixVersion
+  )
+  .defaultAxes(VirtualAxis.jvm)
+  .jvmPlatform(rulesCrossVersions)
 
-lazy val input = project.settings(
-  publish / skip := true
-)
+lazy val input = projectMatrix
+  .settings(
+    publish / skip := true
+  )
+  .jvmPlatform(
+    scalaVersions = rulesCrossVersions,
+    settings = Seq(
+      scalacOptions += "-P:semanticdb:synthetics:on"
+    )
+  )
+  .defaultAxes(VirtualAxis.jvm)
+  .jvmPlatform(scalaVersions = rulesCrossVersions :+ scala3Version)
 
-lazy val output = project.settings(
-  publish / skip := true
-)
+lazy val output = projectMatrix
+  .settings(
+    publish / skip := true
+  )
+  .defaultAxes(VirtualAxis.jvm)
+  .jvmPlatform(scalaVersions = rulesCrossVersions :+ scala3Version)
 
-lazy val tests = project
+lazy val testsAggregate = Project("tests", file("tests/aggregate"))
+  .aggregate(tests.projectRefs: _*)
+
+lazy val tests = projectMatrix
   .settings(
     publish / skip := true,
     libraryDependencies += "ch.epfl.scala" % "scalafix-testkit" % V.scalafixVersion % Test cross CrossVersion.full,
-    Compile / compile :=
-      (Compile / compile).dependsOn(input / Compile / compile).value,
     scalafixTestkitOutputSourceDirectories :=
-      (output / Compile / unmanagedSourceDirectories).value,
+      resolveByInputAxis(output, Compile / unmanagedSourceDirectories).value,
```

Comment:
https://github.com/scalacenter/sbt-scalafix/pull/219/files#diff-5103eedd8d68b8d7d72de5624045280ad313a43e12e44bf6966e591963db3f62R39

Reply:
feeding `output` to an `input`-name method is confusing, maybe the axis and the helper should be called `TestkitDataAxis` / `resolveByTestkitDataAxis`?

Reply:
https://github.com/scalacenter/sbt-scalafix/pull/220